src/NUnitFramework/framework/Constraints/DelayedConstraint.cs

```diff
@@ -35,9 +34,159 @@ namespace NUnit.Framework.Constraints
+            public WithRawPollingInterval PollEvery(int milliSeconds)
+            {
+                _parent._pollingInterval = new Interval(milliSeconds).InMilliseconds;
+                return new WithRawPollingInterval(_parent);
+            }
+        }
+
+        /// <summary>
+        /// Allows only changing the time dimension of delay interval and setting a polling interval of a DelayedConstraint 
+        /// </summary>
+        public class WithRawDelayInterval : DelayedConstraint
+        {
+            private readonly DelayedConstraint _parent;
+
+            /// <summary>
+            /// Creates a new DelayedConstraint.WithRawDelayInterval
+            /// </summary>
+            /// <param name="parent">Parent DelayedConstraint on which delay interval dimension is required to be set</param>
+            public WithRawDelayInterval(DelayedConstraint parent)
+                : base(parent.BaseConstraint, parent._delayInterval, parent._pollingInterval)
+            {
+                _delayInterval = parent._delayInterval;
+                _parent = parent;
+            }
+
+            /// <summary>
+            /// Changes delay interval dimension to minutes
+            /// </summary>
+            public WithDimensionedDelayInterval Minutes
+            {
+                get
+                {
+                    _parent._delayInterval = _parent._delayInterval.InMinutes;
+                    return new WithDimensionedDelayInterval(_parent);
+                }
+            }
+
+            /// <summary>
+            /// Changes delay interval dimension to seconds
+            /// </summary>
+            public WithDimensionedDelayInterval Seconds
+            {
+                get
+                {
+                    _delayInterval = _delayInterval.InSeconds;
+                    return new WithDimensionedDelayInterval(this);
+                }
+            }
+
+            /// <summary>
+            /// Changes delay interval dimension to milliseconds
+            /// </summary>
+            public WithDimensionedDelayInterval MilliSeconds
+            {
+                get
+                {
+                    _delayInterval = _delayInterval.InMilliseconds;
+                    return new WithDimensionedDelayInterval(this);
+                }
+            }
+
+            /// <summary>
+            /// Set polling interval, in milliseconds
+            /// </summary>
+            /// <param name="milliSeconds">A time interval, in milliseconds</param>
+            /// <returns></returns>
+            public WithRawPollingInterval PollEvery(int milliSeconds)
+            {
+                _parent._pollingInterval = new Interval(milliSeconds).InMilliseconds;
+                return new WithRawPollingInterval(_parent);
+            }
+        }
+
         // TODO: Needs error message tests
         private Interval _delayInterval;
-        private readonly int _pollingInterval;
+        private Interval _pollingInterval;
```

Comment:
These private Intervals are of course accessible in the nested classes but it may be clearer to future devs if we make this protected. Could also be a protected automatic Property.